can/interfaces/remote/events.py

```diff
@@ -0,0 +1,451 @@
+"""
+Different events can be sent and transmitted over the network connection.
+
+Examples:
+  * Messages
+  * Exceptions
+  * Transmit success
+  * Transmit failure
+  * ...
+"""
+
+import struct
+import logging
+import can
+
+logger = logging.getLogger(__name__)
+
+
+class BaseEvent(object):
+    """Events should inherit this class."""
+
+    def encode(self):
+        """Convert event data to bytes.
+
+        :return:
+            Bytestring representing the event data.
+        """
+        return b''
```

Comment:
I'm taking advantage of the default when the event does not contain any data. Since it is not a public API I think it makes sense to have a default behavior.